evm/v0.5/contracts/dev/PrepaidAggregator.sol

```diff
@@ -404,42 +451,48 @@ contract PrepaidAggregator is AggregatorInterface, Ownable, WithdrawalInterface
     _;
   }
 
-  modifier onlyIfMinAnswersReceived(uint32 _id) {
+  modifier ifMinAnswersReceived(uint32 _id) {
     if (rounds[_id].details.answers.length >= rounds[_id].details.minAnswers) {
       _;
     }
   }
 
-  modifier onlyIfMaxAnswersReceived(uint32 _id) {
+  modifier ifMaxAnswersReceived(uint32 _id) {
     if (rounds[_id].details.answers.length == rounds[_id].details.maxAnswers) {
       _;
     }
   }
 
-  modifier onlyIfAcceptingAnswers(uint32 _id) {
+  modifier onlyWhenAcceptingAnswers(uint32 _id) {
     require(rounds[_id].details.maxAnswers != 0, "Max responses reached for round");
     _;
   }
 
-  modifier onlyOnNewRound(uint32 _id) {
+  modifier ifNewRound(uint32 _id) {
     if (_id == currentRound.add(1)) {
       _;
     }
   }
 
-  modifier onlyIfDelayedOrOwner(uint32 _id) {
+  modifier ifDelayedOrOwner(uint32 _id) {
     uint256 lastStarted = oracles[msg.sender].lastStartedRound;
-    if (_id > lastStarted + restartDelay || isOwner()) {
+    if (_id > lastStarted + restartDelay || lastStarted == 0 || isOwner()) {
       _;
     }
   }
 
   modifier onlyValidRoundId(uint32 _id) {
     require(_id == currentRound || _id == currentRound.add(1), "Must report on current round");
-    require(rounds[_id.sub(1)].updatedTimestamp > 0 || _id == 1, "Cannot bump round until previous round has an answer");
+    require(_id == 1 || answered(_id.sub(1)) || timedOut(_id.sub(1)), "Cannot bump round until previous round has an answer");
```

Comment:
`timedOut` already decrements the `_id` when checking the round's updatedTimestamp. Should we be decrementing it again here in the last case?

Reply:
Yes. We want to check if the previous round has timed out, so we pass in the id one round back. Then to find out if that round is timed out, we have to look at the timestamp one more round back and compare it to the current time. More concretely: round 3 is the first round that could be forced from timeout, because you need a round 1 timestamp to see if round 2 timed out.